system/loggerd/deleter.py

```diff
@@ -20,7 +29,19 @@ def deleter_thread(exit_event):
     if out_of_percent or out_of_bytes:
       # remove the earliest directory we can
       dirs = sorted(listdir_by_creation(ROOT), key=lambda x: x in DELETE_LAST)
-      for delete_dir in dirs:
+
+      # skip deleting most recent N preserved segments (and their prior segment)
+      preserved_dirs = []
+      for n, d in enumerate(filter(has_preserve_xattr, reversed(dirs))):
+        if n == PRESERVE_COUNT:
+          break
+        date_str, _, seg_num = d.rpartition("--")
+
+        # preserve segment and its prior
+        preserved_dirs.append(d)
+        preserved_dirs.append(f"{date_str}--{int(seg_num) - 1}")
```

Comment:
this should be handled the same way we handle `DELETE_LAST`, just move them to the back in the sorting. we need a test case that catches the way this is written now; this breaks if you have less than 5 segments of storage

Reply:
sorting should go:
* boot/crash
* preserved segments, sorted by creation
* rest of segments, sorted by creation